synapse/lib/agenda.py

```diff
@@ -809,7 +809,20 @@ async def _scheduleLoop(self):
                     logger.warning(mesg,
                                    extra={'synapse': {'iden': appt.iden, 'name': appt.name}})
                 else:
-                    await self._execute(appt)
+                    try:
+                        await self._execute(appt)
+                    except Exception as e:
+                        extra = {'iden': appt.iden, 'name': appt.name, 'user': appt.creator, 'view': appt.view}
+                        user = self.core.auth.user(appt.creator)
+                        if user is not None:
+                            extra['username'] = user.name
+                        if isinstance(e, s_exc.SynErr):
+                            mesg = e.get('mesg', str(e))
+                        else:  # pragma: no cover
+                            mesg = str(e)
+                        logger.exception(f'Agenda error running appointment {appt.iden} {appt.name}: {mesg}',
```

Comment:
Last I checked, general guidance is to not use f-strings with logging functions. I think it's because the string substitutions happen every time with f-strings where if you use the format specifiers, the substitutions only happen if the message actually gets logged.